### DISCUSSION TITLE
I can't figure out how to generate an API key.

### DISCUSSION THREAD

**Discussion by 2gn** [khoj-ai/khoj#681] [Q&A] [answered]
From the desktop app I opened http://localhost:42110/config to generate an API key, but I can't see anything related to the API.
How do I generate one?
![khoj says this](https://github.com/khoj-ai/khoj/assets/101851090/bc365d2a-998a-4a1e-8343-1ff95ae8287e)
![config page does not mention anything related to the API](https://github.com/khoj-ai/khoj/assets/101851090/283bcf5c-e962-4fc7-8f96-7d3dd084de6b)

**Comment by 2gn** [comment, marked as answer]
Turned out I was running Khoj in anonymous mode. It's specified inside docker-compose.yml.

As discussed in #674, when running khoj in anonymous-mode you don't need to set API key.

My problem was that I configured http"s"://localhost:42110 as the server URL, but it doesn't work since it doesn't support SSL by default.

**Reply by debanjum** [reply]
Yeah, API key isn't required in anonymous mode. Maybe we should mention this in the docs, as it seems like multiple folks are getting confused by it
